src/control/server/storage/provider.go

```diff
@@ -288,11 +288,11 @@ func (p *Provider) FormatBdevTiers() (results []BdevTierFormatResult) {
 			continue
 		}
 
-		p.RLock()
+		p.Lock()
 		req.BdevCache = &p.bdevCache
```

Comment:
I found myself wondering if perhaps the spdk operations should be locking at a lower level, for example in the bdev provider itself. At this level it's hard to tell what exactly the conflict was. Maybe I'm wrong but on re-visiting this code, this lock appears to be originally intended to protect the bdev cache.

I'm not totally against expanding its purpose, but it seems like "spdk operations can conflict" is bdev provider specific knowledge.

Reply:
For the same reason as protecting the bdev cache, it needs to be at this level (storage provider) in order to persist across daos_admin process reincarnations. Locking in the bdev provider backend wouldn't protect against multiple invocations from the storage provider level. Make sense?

Reply:
Ah, understood.